rust/operator-binary/src/discovery.rs

```diff
@@ -21,12 +21,16 @@ pub fn build_discovery_configmap(
     controller: &str,
     namenode_podrefs: &[HdfsPodRef],
     resolved_product_image: &ResolvedProductImage,
-) -> OperatorResult<ConfigMap> {
+) -> Result<ConfigMap, crate::hdfs_controller::Error> {
```

Comment:
**nit:** Snafu proposes to have an error type per module (https://docs.rs/snafu/latest/snafu/guide/philosophy/index.html#many-error-types).

Reply:
Yeah we currently have a mess, I want that as well. We need to address that at some point of time, I just fear this is out of scope for this PR

Reply:
FYI, #454 brings in some Snafu enums

Reply:
Awesome, thanks!